### ISSUE TITLE
Converting Google ROM To microG Is Problems - Not Debloating!

### ISSUE THREAD

**Issue by 0pLuS0** [microg/android_packages_apps_GmsCore#699]
I'm starting to run into more issues then I realized when I started this!

It seems like microG only works best on a AOSP ROM, but when you have a Google ROM and you want to convert it, the only way about this is first installing the Google ROM, letting it run and various applications that depend on GSF, then DEBLOAT out GSF and Gapps, then replace with microG.

But what I am doing is ripping out GSF & Play Services from a Google ROM and then building it in SuperR's Kitchen to be flashed in TWRP and now I run into some things not working, which also appears to be the System itself based around GSF!

Here is one issue I am up against, trying to get the ```Share``` button to work in Gallery Applications and they won't!

So I filed a bug report with SimpleGallery in hopes of figuring this out.

https://github.com/SimpleMobileTools/Simple-Gallery/issues/1257

This is on OxygenOS 5.1.7 for the OP5T converted to using microG.

Also the OnePlus Gallery share button doesn't work either, then I took the LineageOS Gallery out of their ROM and thought it would work, same issue, the LOS Gallery share button does nothing.

SO I am not sure how this can be supported in microG when trying to convert a Google ROM to microG, not debloating it after it's been installed....

OH THE PAIN! :(

**Comment by 0pLuS0** [comment]
I've come to realize in my microG adventures that if have a Google ROM like OxygenOS from OnePlus and you want everything to work, or as close as possible, you need to run the ROM first, then debloat of Google and then install microG.

But with tools like SuperR's Kitchen, I have been removing all the Google and working on OxygenOS as a stripped down microG you flash in TWRP like any other ROM, problem is, because the ROM is based around Google and it's not starting with the Google Framework at first, then various things in Google ROMs I've noticed don't work!

Here's the REAL KICKER as I like to say! What I found odd in all of this is that once you run a Google ROM, then debloated of Google and install microG, this so called Google dependent ROM and various apps that need the Google Framework will continue running. Now you might be thinking they will also run with microG, no, I am talking about the ROM and various ROM features/functions and it's apps many of them will not properly work unless it's run under the Google Framework & Services, and then once they system and various apps have run once, you can then replace Google with microG and they will still continue to work!

But convert a Google ROM into a microG with a tool like SuperR's Kitchen, where it is flashed in TWRP to install, then it boots for the first time with ONLY microG, guess what, these same ROM features, functions and apps many of them will not work!

SO my question is, WHY can't microG be made to work from the beginning, since it's obvious that only the Google Framework & Services need to be used once, then microG can be introduced?

I would of thought microG could be made to work like this, but I am seeing it is not, when someone is using it on a Google ROM.

I hope those working in microG can make this happen!

THANKS

**Comment by 0pLuS0** [comment]
I've come to realize in my microG adventures that with a Google ROM like OxygenOS from OnePlus and you want everything to work, or as close as possible, you need to run the OxygenOS ROM first, then debloat Google, then install microG.

But with tools like SuperR's Kitchen, I have been removing all the Google and working on OxygenOS as a stripped down microG ROM you flash in TWRP like any other ROM, problem is, because the ROM is based around Google and it's not starting with the Google Framework at first, then various things in Google ROMs I've noticed don't work!

Here's the REAL KICKER as I like to say! 

What I found odd in all of this is that once you run a Google ROM, then debloat out the Google, next install microG, this so called Google dependent ROM and various apps that need the Google Framework will continue running. Now you might be thinking this does work with microG, no, I am talking about the ROM and various ROM features/functions and it's apps, many of them will not properly work unless it's run under the Google Framework & Services first, and then once the system and various apps have run once, then you can then replace Google with microG and they will still continue to work!

Understand?

But start out converting a Google ROM into a microG ROM with a tool like SuperR's Kitchen, where it is flashed in TWRP to install, then boot it with ONLY microG, guess what, these same ROM features, functions and apps many of them will not work!

SO my question is, WHY can't microG be made to work from the beginning, since it's obvious that only the Google Framework & Services need to be used once, then microG can be introduced?

Why can't microG seem to Kick Start the system and apps to make it all work from the beginning as I'm asking here? I would of thought microG could be made to work like this, but I am seeing it is not when someone is using it on a Google ROM.

So BOTTOM LINE, you have a Google ROM, then you need to install and run it first, then debloat Google, then install microG and it all works!

Or you have a AOSP ROM and you can use microG from the start!

I want to be able to take a Google ROM like OxygenOS, rip out the Google and make it into a microG ROM with SuperR's Kitchen, then flash it in TWRP and make this work, starting out more as a microG ROM, which I also thought was a point of the microG project, to work anyway you want to apply it, but that's not true.

I hope those working in microG can make this happen!

THANKS
